runner/local/queueing_runner.go

```diff
@@ -0,0 +1,339 @@
+package local
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/scootdev/scoot/runner"
+)
+
+// TODO(dbentley): should this be in queue.go
+// TODO(dbentley): should this whole package move to runner/runners?
+
+const QueueFullMsg = "No resources available. Please try later."
+const UnknownRunIdMsg = "Unknown run id."
+const RequestIsRunning = "Run %s is still running, please Abort it first."
+
+// QueueingRunner manages a queue of commands.
+// One request is run at a time, via an underlying delegate runner
+// (We expect, for now, the delegate runner will ultimately be a simpleRunner
+// QueueingRunner will reject requests if the queue is full (i.e., its length == its capacity)
+type QueueingRunner struct {
+	// channel for methods to send communication to eventLoop()
+	reqCh chan interface{}
+
+	// delegate that will run (we expect it to ultimately be a simpleRunner)
+	delegate    runner.Runner
+	capacity    int
+	nextRunId   int // used to assign unique runIds to the run requests
+	runnerAvail bool
+
+	// Runs are in one of three meta-states in QueueingRunner:
+	// *) queued
+	// *) errored (we couldn't delegate them)
+	// *) delegated
+
+	// q holds queued commands (and assigned ids)
+	q []commandAndId
+
+	// errored holds info for runs that have errored.
+	// We weren't able to send them to the delegate (either got an error from Run()
+	// or they were aborted before running
+	errored map[runner.RunId]runner.ProcessStatus
+
+	// delegated runs are stored in both qToDel and delToQ. Together, these
+	// form a BiMap (but Go doesn't make that easy), so we have two maps.
+	// qToDel maps IDs in our namespace to IDs in the delegate's namespace
+	// delToQ is vice versa
+	qToDel map[runner.RunId]runner.RunId
+	delToQ map[runner.RunId]runner.RunId
+}
+
+// NewQueueingRunner creates a new QueueingRunner.
+// delegate will be used to run.
+// capacity is the maximum entries to hold in the queue
+// runnerAvailableCh should send a struct whenever the runner is available
```

Comment:
change to 
// the delegate runner should send a struct{}{} on runnerAvailableCh when the delegate runner becomes available